core/src/main/java/org/geysermc/geyser/registry/populator/CustomItemRegistryPopulator.java

```diff
@@ -271,6 +271,19 @@ private static void setupBasicItemInfo(int maxDamage, int stackSize, boolean dis
                 .build());
         componentBuilder.putCompound("minecraft:display_name", NbtMap.builder().putString("value", customItemData.displayName()).build());
 
+        // Add a geyser tag to the item, allowing molang to be used
+        setItemTag(componentBuilder, "geyser_custom");
+
+        // Add other defined tags to the item
+        Set<String> tags = customItemData.tags();
+        if (tags != null) {
+            for (String tag : tags) {
+                if (!tag.isEmpty()) {
+                    setItemTag(componentBuilder, tag);
```

Comment:
At first glance it looks like `setItemTag` overrides existing tags on the componentBuilder

Reply:
Should also do a null check on `tag` here